internal/controllers/nmc_reconciler.go

```diff
@@ -167,7 +167,7 @@ func (r *NMCReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 
 	// removing label of loaded kmods
 	if !r.nodeAPI.IsNodeSchedulable(&node, nil) {
```

Comment:
using IsNodeSchedulable with nil will always return true, so we can simply delete this "if" clause. But i suggest that we actually collect the modules that are not schedulable on the node at line 139, and then we just delete their labels at this line (169).  This way it also means that we don't need the whole node-taints-tolerations loop again in the RemoveNodeReadyLabels function

Reply:
good idea